src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivityService.kt

```diff
@@ -63,7 +63,9 @@ class ActivityService(
   private fun mapActivityModel(request: CreateActivityRequest): CourseActivity {
     val prison = findPrisonOrThrow(request.prisonId)
 
-    val location = findLocationInPrisonOrThrow(request.internalLocationId, request.prisonId)
+    val location = request.internalLocationId
+      ?.let { findLocationInPrisonOrThrow(request.internalLocationId, request.prisonId) }
+      ?: findCswapLocationInPrisonOrThrow(request.prisonId)
```

Comment:
The main change - default CSWAP if requested null location (and same for updates below)